script.logviewer/resources/lib/utils.py

```diff
@@ -1,39 +1,58 @@
 # -*- coding: utf-8 -*-
 
+import logging
 import sys
 
+import xbmc
 import xbmcaddon
 
 PY3 = sys.version_info.major >= 3
 
 ADDON = xbmcaddon.Addon()
 ADDON_PATH = ADDON.getAddonInfo("path")
 ADDON_NAME = ADDON.getAddonInfo("name")
+ADDON_ID = ADDON.getAddonInfo("id")
 
 if PY3:
     def translate(text):
         return ADDON.getLocalizedString(text)
+
+    def encode(s):
+        return s.encode("utf-8")
+
+    def decode(s):
+        return s.decode("utf-8")
 else:
     ADDON_PATH = ADDON_PATH.decode("utf-8")
 
     def translate(text):
         return ADDON.getLocalizedString(text).encode("utf-8")
 
+    def encode(s):
+        return s
+
+    def decode(s):
+        return s
+
 
 def get_setting(setting):
     return ADDON.getSetting(setting)
 
 
-def get_boolean(setting):
+def get_boolean_setting(setting):
```

Comment:
Kodi provides methods to get settings as bools in newest releases (Leia): `getSettingBool`, `getSettingInt`, etc, see: https://codedocs.xyz/xbmc/xbmc/group__python__xbmcaddon.html#gaf0205dd5b96354f30a39fbf35599e681

Maybe you can do some sort of runtime version detection and use those instead

Reply:
I would prefer to keep these ones so I don't have to do compatibility checks.